lang/cs/Org.Apache.REEF.Network/Elastic/Operators/Logical/ElasticOperator.cs

```diff
@@ -0,0 +1,591 @@
+            {
+                GetOperatorConfiguration(ref serializedOperatorsConfs, taskId);
+
+                if (_next != null)
+                {
+                    _next.GetTaskConfiguration(ref serializedOperatorsConfs, taskId);
+                }
+            }
+            else
+            {
+                throw new IllegalStateException("Operator needs to be finalized before getting tasks configuration.");
+            }
+        }
+
+        /// <summary>
+        /// Whether this operator is ready to be scheduled by the task set manager.
+        /// </summary>
+        /// <returns>True if the operator is ready to be scheduled</returns>
+        internal bool CanBeScheduled()
+        {
+            bool canBeScheduled = _topology.CanBeScheduled();
+
+            if (canBeScheduled && _next != null)
+            {
+                return _next.CanBeScheduled();
+            }
+
+            return canBeScheduled;
+        }
+
+        /// <summary>
+        /// Utility method gathering the set of master task ids of the operators in the current pipeline.
+        /// </summary>
+        /// <param name="masterTasks">The id of the master tasks of the current and successive operators</param>
+        internal virtual void GatherMasterIds(ref HashSet<string> masterTasks)
+        {
+            if (_operatorFinalized != true)
+            {
+                throw new IllegalStateException("Operator need to be build before gathering information.");
+            }
+
+            masterTasks.Add(Utils.BuildTaskId(Stage.StageName, MasterId));
+
+            if (_next != null)
+            {
+                _next.GatherMasterIds(ref masterTasks);
+            }
+        }
+
+        /// <summary>
+        /// Log the final statistics of the operator.
+        /// This is called when the pipeline execution is completed.
+        /// </summary>
+        internal virtual string LogFinalStatistics()
+        {
+            var str = LogInternalStatistics();
+
+            if (_next != null)
+            {
+                str += _next.LogFinalStatistics();
+            }
+
+            return str;
+        }
+
+        /// <summary>
+        /// Appends the message type to the configuration.
+        /// </summary>
+        /// <param name="operatorType">The type of the messages the operator is configured to accept</param>
+        /// <param name="confBuilder">The configuration builder the message type will be added to</param>
+        protected void SetMessageType(Type operatorType, ref ICsConfigurationBuilder confBuilder)
+        {
+            if (operatorType.IsGenericType)
+            {
+                var genericTypes = operatorType.GenericTypeArguments;
+                var msgType = genericTypes[0];
+                confBuilder.BindNamedParameter<OperatorParameters.MessageType, string>(
+                    GenericType<OperatorParameters.MessageType>.Class, msgType.AssemblyQualifiedName);
```

Comment:
maybe
```C#
confBuilder.BindStringNamedParam<OperatorParameters.MessageType>(msgType.AssemblyQualifiedName);
```
(don't remember if we need to reassign the `confBuilder`)

Reply:
Underneath it is calling `BindNamedParameter` so we are good. Fixed.